spring-data-mongodb/src/main/java/org/springframework/data/mongodb/gridfs/ReactiveGridFsTemplate.java

```diff
@@ -216,9 +228,7 @@ public ReactiveGridFsTemplate(DataBufferFactory dataBufferFactory, ReactiveMongo
 
 		Assert.notNull(file, "GridFSFile must not be null!");
 
-		return Mono.fromSupplier(() -> {
-			return new ReactiveGridFsResource(file, getGridFs().downloadToPublisher(file.getId()), dataBufferFactory);
-		});
+		return Mono.fromSupplier(() -> new ReactiveGridFsResource(file, this.doGetBucket(), dataBufferFactory));
```

Comment:
Let's rewrite this method to `doGetBucket().map(it ->new ReactiveGridFsResource(…))` instead of introducing a new constructor.

Reply:
Fixed!